src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java

```diff
@@ -91,4 +96,50 @@ public String deleteSupplier(@PathVariable("procID") final Integer procId,
 
     return "OK";
   }
+
+  @GetMapping("/{eventID}/documents")
+  public Collection<DocumentSummary> getDocumentSummaries(
+      @PathVariable("procID") final Integer procId, @PathVariable("eventID") final String eventId,
+      final JwtAuthenticationToken authentication) {
+
+    var principal = getPrincipalFromJwt(authentication);
+    log.info("getDocumentSummaries invoked on behalf of principal: {}", principal);
+
+    return procurementEventService.getDocumentSummaries(procId, eventId);
+  }
+
+  @PutMapping("/{eventID}/documents")
+  public DocumentSummary uploadDocument(@PathVariable("procID") final Integer procId,
+      @PathVariable("eventID") final String eventId,
+      @RequestParam("data") final MultipartFile multipartFile,
+      @RequestParam("audience") final String audience,
```

Comment:
The generated enum (`DocumentAudienceType`) has a `fromValue` method annotated with `@JsonCreator` - so I think you can go straight to that type

Reply:
I had that originally - but it only worked if the param was in upper case - e.g. `BUYER` (even though it has lower case constructors generated from the API) - I'll give it another try though